File: Assignment11/C/Program2.c
// PS - Accept N number from user and accept one another number as NO, return index of 1st occurance of that number

#include<stdio.h>
#include<stdlib.h>

int CheckNum(int Arr[], int iLength, int No) {
    int iCnt = 0;
    int index = 0;

    for(iCnt = 0; iCnt < iLength; iCnt++){
        if(Arr[iCnt] == No) {
            index = iCnt;
            break;
        }
    }

    return index;
}

int main() {
    int iSize = 0, iValue = 0, iCnt = 0;
    int *ptr = NULL;
    int iRet = 0;

    printf("Enter number of elements that you want to store in array \n");
    scanf("%d", &iSize);

    printf("Enter number which you want to check \n");
    scanf("%d", &iValue);

    ptr = (int *)malloc(sizeof(int) * iSize);

    printf("Enter elemets in array \n");
    for(iCnt = 0; iCnt < iSize; iCnt++) {
        scanf("%d", &ptr[iCnt]);
    }

    iRet = CheckNum(ptr, iSize, iValue);

    printf("1st occurance of %d is %d", iValue, iRet);

    free(ptr);

    return 0;
}

File: Assignment36/Java/Program2.java
// PS - write java program which accept N numbers from user and accept one more number and return index of 1st occurance of that number
import java.util.*;

class ArrayX
{
    public int CheckNumIndex(int Arr[], int No)
    {
        for(int iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            if(Arr[iCnt] == No)
            {
                return iCnt;
            }
        }
        return -1;
    }
}

class Program2
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number of elements");
        int iSize = sobj.nextInt();

        int Arr[] = new int[iSize];

        System.out.println("Enter elements of array");
        for(int iCnt = 0; iCnt < iSize; iCnt++)
        {
            Arr[iCnt] = sobj.nextInt();
        }

        System.out.println("Enter number that you want to search");
        int No = sobj.nextInt();

        ArrayX aobj = new ArrayX();
        int iret = aobj.CheckNumIndex(Arr, No);
        
        if(iret == -1)
        {
            System.out.println("number is not present");
        } else
        {
            System.out.println("1st occurance is at index :"+iret);
        }
    }
}